src/components/menu/menu-item/menu-item.component.tsx

```diff
@@ -58,13 +127,15 @@ const MenuItem = ({
     handleKeyDown: handleSubmenuKeyDown,
     shiftTabPressed,
   } = submenuContext;
-  const ref = useRef(null);
+  const ref = useRef<HTMLAnchorElement & HTMLButtonElement & HTMLDivElement>(
+    null
+  );
   const focusFromMenu = focusId === menuItemId.current;
   const focusFromSubmenu = submenuFocusId
     ? submenuFocusId === menuItemId.current
     : undefined;
-  const inputRef = useRef(null);
-  const inputIcon = useRef(null);
+  const inputRef = useRef<HTMLInputElement | null | undefined>(null);
```

Comment:
I'm not really sure what you mean tbh, can you provide a code example? The current implementation is necessary as the `querySelector` used to assign the input and icon refs may return an element or null but tsc thinks the main `ref` may not have been initialised and that therefore would cause the other refs to be `undefined`. I can update it to remove the additonal type if I go with something like below, let me know if you have a different approach

```
const inputRef = useRef<HTMLInputElement | null>(null);
  const inputIcon = useRef<HTMLSpanElement | null>(null);

  inputIcon.current = ref.current ? ref.current?.querySelector(
    "[data-element='input-icon-toggle']"
  ) : null;
  inputRef.current = ref.current ? ref.current?.querySelector("[data-element='input']") : null;
  ```

Reply:
I'm happy to go with this example 👍🏼